client/src/components/tree-select/component.tsx

```diff
@@ -201,76 +193,44 @@ const TreeSelect: React.FC<TreeSelectProps> = ({
     [checkedKeys, onChange, options],
   );
 
-  const Icon = () => (
-    <span className="absolute inset-y-0 right-0 flex items-center pr-2 pointer-events-none">
-      {isOpen ? (
-        <ChevronUpIcon className="h-4 w-4 text-gray-900" aria-hidden="true" />
-      ) : (
-        <ChevronDownIcon className="h-4 w-4 text-gray-900" aria-hidden="true" />
-      )}
-    </span>
-  );
-
   // Current selection
   useEffect(() => {
     if (current && current.length) {
       const currentKeys = current.map(({ value }) => value);
-      setSelected(current[0]);
       setSelectedKeys(currentKeys);
       setCheckedKeys(currentKeys);
     }
   }, [current]);
 
   return (
     <div className="relative">
-      {multiple ? (
-        <div
-          className="flex align-center bg-white relative border border-gray-300 rounded-md shadow-sm py-2 pr-10 pl-3 cursor-pointer"
-          onClick={handleToggleOpen}
-        >
-          <div className="flex flex-wrap">
-            {currentOptions &&
-              !!currentOptions.length &&
-              currentOptions.slice(0, maxBadges).map((option) => (
-                <Badge
-                  key={option.value}
-                  className="text-sm m-0.5"
-                  data={option}
-                  onClick={handleRemoveBadget}
-                  removable
-                >
-                  {option.label}
-                </Badge>
-              ))}
-            {currentOptions && currentOptions.length > maxBadges && (
-              <Badge className="text-sm m-0.5">
-                {currentOptions.length - maxBadges} more selected
-              </Badge>
-            )}
-            {(!currentOptions || currentOptions.length === 0) && (
-              <span className="inline-block truncate">
-                {placeholder && <span className="text-gray-300 text-sm">{placeholder}</span>}
-              </span>
-            )}
-          </div>
-          <Icon />
-        </div>
-      ) : (
-        <button
-          type="button"
-          className="bg-white relative w-full flex align-center border border-gray-300 rounded-md shadow-sm pl-3 pr-10 py-2 text-left
-          focus:outline-none focus:ring-1 focus:ring-green-700 focus:border-green-700 text-sm cursor-pointer"
-          onClick={handleToggleOpen}
-        >
-          <span className="inline-block truncate">
-            {selected ? (
-              <span className="font-medium">{selected.label}</span>
-            ) : (
-              <span className="text-gray-300">{placeholder}</span>
-            )}
-          </span>
-          <Icon />
-        </button>
+      {theme === 'primary' && (
+        <TreeSelectButtonPrimary
```

Comment:
It's they have different functionality or behavior they should be different components. In case the props, functionality, and everything is the same but only change the theme, it should be only controlled by `classnames`.